Tests/GRPCTests/GRPCReflectionServiceTests/ReflectionServiceV1AlphaIntegrationTests.swift

```diff
@@ -0,0 +1,349 @@
+import NIOPosix
+import SwiftProtobuf
+import XCTest
+
+@testable import GRPCReflectionService
+
+final class ReflectionServiceV1AlphaIntegrationTests: GRPCTestCase {
+  private var server: Server?
+  private var channel: GRPCChannel?
+  private let protos: [Google_Protobuf_FileDescriptorProto] = makeProtosWithDependencies()
+  private let independentProto: Google_Protobuf_FileDescriptorProto = generateFileDescriptorProto(
+    fileName: "independentBar",
+    suffix: "5"
+  )
+
+  private func setUpServerAndChannel() throws {
+    let reflectionServiceProvider = try ReflectionService(
+      fileDescriptorProtos: self.protos + [self.independentProto],
+      version: .v1Alpha
+    )
+
+    let server = try Server.insecure(group: MultiThreadedEventLoopGroup.singleton)
+      .withServiceProviders([reflectionServiceProvider])
+      .withLogger(self.serverLogger)
+      .bind(host: "127.0.0.1", port: 0)
+      .wait()
+    self.server = server
+
+    let channel = try GRPCChannelPool.with(
+      target: .hostAndPort("127.0.0.1", server.channel.localAddress!.port!),
+      transportSecurity: .plaintext,
+      eventLoopGroup: MultiThreadedEventLoopGroup.singleton
+    ) {
+      $0.backgroundActivityLogger = self.clientLogger
+    }
+
+    self.channel = channel
+  }
+
+  override func tearDown() {
+    if let channel = self.channel {
+      XCTAssertNoThrow(try channel.close().wait())
+    }
+    if let server = self.server {
+      XCTAssertNoThrow(try server.close().wait())
+    }
+
+    super.tearDown()
+  }
+
+  func testFileByFileName() async throws {
+    try self.setUpServerAndChannel()
+    let client = Grpc_Reflection_V1alpha_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.fileByFilename = "bar1.proto"
+      }
+    )
+    serviceReflectionInfo.requestStream.finish()
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    guard let message = try await iterator.next() else {
+      return XCTFail("Could not get a response message.")
+    }
+
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(
+        serializedData: (message.fileDescriptorResponse
+          .fileDescriptorProto[0])
+      )
+
+    XCTAssertEqual(receivedFileDescriptorProto.name, "bar1.proto")
+    XCTAssertEqual(receivedFileDescriptorProto.service.count, 1)
+
+    guard let service = receivedFileDescriptorProto.service.first else {
+      return XCTFail("The received file descriptor proto doesn't have any services.")
+    }
+    guard let method = service.method.first else {
+      return XCTFail("The service of the received file descriptor proto doesn't have any methods.")
+    }
+    XCTAssertEqual(method.name, "testMethod1")
+    XCTAssertEqual(message.fileDescriptorResponse.fileDescriptorProto.count, 4)
+  }
```

Comment:
I don't love that we've essentially copied-and-pasted all the tests. It's going to be difficult to maintain going forward.

I think we can be a bit smarter here by having each case iterate over the two versions and having some convenience functions to switch between v1 and v1 alpha. This obviously won't compile but hopefully gives an idea of what we can do:

```suggestion
  func testFileByFileName() async throws {
    for version in versions {
      try self.setUpServerAndChannel(version: version)

      let request = V1Request.with { 
        $0.host = "127.0.0.1"
        $0.fileByFilename = "bar1.proto"
      }

      let response: V1Response?

      switch version {
      case .v1:
        let client = V1Client(channel: self.channel!)
        let call = client.makeServerReflectionInfoCall()
        try await call.send(request)
        call.finish()
        var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
        response = try await iterator.next()
      case .v2:
        let client = V1Client(channel: self.channel!)
        let call = client.makeServerReflectionInfoCall()
        try await call.send(V1AlphaRequest(request)) // convert from v1 to alpha
        call.finish()
        response = try await iterator.next().map { V1AlphaResponse($0) } // convert from v1 to alpha
      }

      guard let message = response else {
        return XCTFail("Could not get a response message.")
      }

      let receivedFileDescriptorProto =
        try Google_Protobuf_FileDescriptorProto(
          serializedData: (message.fileDescriptorResponse
            .fileDescriptorProto[0])
        )

      XCTAssertEqual(receivedFileDescriptorProto.name, "bar1.proto")
      XCTAssertEqual(receivedFileDescriptorProto.service.count, 1)

      guard let service = receivedFileDescriptorProto.service.first else {
        return XCTFail("The received file descriptor proto doesn't have any services.")
      }
      guard let method = service.method.first else {
        return XCTFail("The service of the received file descriptor proto doesn't have any methods.")
      }
      XCTAssertEqual(method.name, "testMethod1")
      XCTAssertEqual(message.fileDescriptorResponse.fileDescriptorProto.count, 4)
    }

```